### PR TITLE
OKTA-346874 restructure projects

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
- Restructured projects so platform specific Okta code is contained in library projects and demo applications are in projects that reference the libraries.
- All files in the Samples/xamarin subdirectory are from the default Xamarin Forms template with minimal Okta specific code for demonstration purposes.

No logic changes were made, just separated code relevant to the demo apps from code relevant to the Okta libraries.

### CODE REVIEW

File: Okta.Xamarin/Okta.Xamarin.Android/OktaLoginCallbackInterceptorActivity.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using Android.App;
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin.Android/OktaLoginCallbackInterceptorActivity.cs:1]: Missing header

File: Okta.Xamarin/Okta.Xamarin.iOS/OktaAppDelegate.cs [hunk 2]
```diff
@@ -1,8 +1,4 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-
+﻿
```
Comment (laura-rodriguez) [inline Okta.Xamarin/Okta.Xamarin.iOS/OktaAppDelegate.cs:1]: Missing header

**Review by laura-rodriguez — APPROVED** [review]
A few missing headers, but not a blocker at all. LGTM.

**Comment by bryanapellanes-okta** [comment]
> A few missing headers, but not a blocker at all. LGTM.

I'll put the headers into the release branch to avoid merge conflict.  Thanks!

**Pull request merged**